### PR TITLE
home-assistant: make hass cloud integration work

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

I'd like to use my "Amazon ~~~spy-device~~~ Echo" to control home assistant.

I haven't actually tried to use it to control HA yet but at least it's connecting.

Should I separate out the pycryptodome override into its own file?

Cc: @dotlambda @FRidh 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -317,6 +317,14 @@ in {
 
   pycryptodome = callPackage ../development/python-modules/pycryptodome { };
 
+  pycryptodome_3_3 = self.pycryptodome.overridePythonAttrs (oldAttrs: rec {
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: There is a policy of having only a single version of each package in `pythonPackages`

File: pkgs/development/python-modules/python-jose-cryptodome/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+buildPythonPackage rec {
+  pname = "python-jose-cryptodome";
+  version = "1.3.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0vl22411k394biizkjqg1ydclbf6fw5hxixzm1g7jlpzx4z6j9lj";
+  };
+
+  propagatedBuildInputs = [ cryptography ecdsa future pycryptodome_3_3 six ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-jose-cryptodome/default.nix]: You can probably just patch `requirements.txt` in `postPatch`.

File: pkgs/development/python-modules/warrant/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  # move to fetchPyPi when https://github.com/capless/warrant/issues/97 is fixed
+  src = fetchFromGitHub {
+    owner  = "capless";
+    repo   = "warrant";
+    rev    = "ff2e4793d8479e770f2461ef7cbc0c15ee784395";
+    sha256 = "0gw3crg64p1zx3k5js0wh0x5bldgs7viy4g8hld9xbka8q0374hi";
+  };
+
+  propagatedBuildInputs = [ boto3 envs python-jose-cryptodome requests ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/warrant/default.nix]: I opened https://github.com/capless/warrant/pull/98 for using python-jose instead.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please rebase on staging, pycryptodome has been updated there.

File: pkgs/development/python-modules/python-jose-cryptodome/default.nix (hunk 2 above)
Comment (Mic92) [inline pkgs/development/python-modules/python-jose-cryptodome/default.nix]: Mhm this could be a big hassle if `python-jose-cryptodome` is mixed with something depending on `pycryptodome`. Then there are two different versions. which should be avoided.

Comment (Mic92) [reply pkgs/development/python-modules/python-jose-cryptodome/default.nix]: Maybe a newer version could be used here? https://github.com/mpdavis/python-jose/blob/master/setup.py#L58

File: pkgs/development/python-modules/python-jose-cryptodome/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ stdenv, buildPythonPackage, fetchPypi
+, cryptography, ecdsa, future, pycryptodome_3_3, six }:
+
+buildPythonPackage rec {
+  pname = "python-jose-cryptodome";
```
Comment (Mic92) [inline pkgs/development/python-modules/python-jose-cryptodome/default.nix]: Also why did you add another `python-jose`?

File: pkgs/development/python-modules/warrant/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/development/python-modules/warrant/default.nix]: You should be able to use that as a patch, which would get rid of all problems.

Comment (dotlambda) [reply pkgs/development/python-modules/warrant/default.nix]: @peterhoeg Did you try applying that patch?

Comment (peterhoeg) [reply pkgs/development/python-modules/warrant/default.nix]: No, haven't touched this yet. This weekend!

**Comment by dotlambda** [comment]
No need to use staging anymore, has been merged into master.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.envs python3.pkgs.envs

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.warrant python3.pkgs.warrant

File: pkgs/development/python-modules/warrant/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  postPatch = ''
+    substituteInPlace requirements.txt \
+      --replace "python-jose-cryptodome>=1.3.2" "python-jose>=2.0.0"
+  '';
+
+  checkInputs = [ mock ];
+
+  propagatedBuildInputs = [ boto3 envs python-jose requests ];
+
+  # all the checks are failing
```
Comment (dotlambda) [inline pkgs/development/python-modules/warrant/default.nix]: Their Travis build fails, too. So this is to be expected.

**Review by dotlambda — APPROVED** [review]
